misc/vault-integration/vault.sh

```diff
@@ -1,55 +1,23 @@
 #!/bin/bash
+set -e
 
 if [[ -z "$VAULT_ADDR" || -z "$VAULT_NAMESPACE" || -z "$VAULT_ROLE" ]]; then
     echo "MISSING REQUIRED VARIABLES: VAULT_ROLE, VAULT_NAMESPACE, VAULT_ADDR"
     exit 1
 fi
-# INSTALLING VAULT
-echo "1. Installing vault..."
+
+echo "Installing vault..."
+
 curl -sL https://releases.hashicorp.com/vault/1.11.1/vault_1.11.1_linux_amd64.zip -o vault1.zip
 unzip -o vault1.zip
 ./vault --version
-echo "================="
-if [[ ! -z "$VAULT_TOKEN" ]]; then
-    echo '1.1 Found VAULT_TOKEN, creating JWT role... '
-    # CREATION
-    # VAULT_TOKEN IS REQUIRED FOR CREATION
-    echo "VAULT_ADDR: ${VAULT_ADDR}"
-    echo "VAULT_NAMESPACE: ${VAULT_NAMESPACE}"
-    echo "VAULT_ROLE: ${VAULT_ROLE}"
 
-    if [[ $STAGE == "prod" ]]; then
-      ENV0_BOUND_AUDIENCE="https://prod.env0.com"
-      AUTH0_BOUND_AUDIENCE="https://env0.auth0.com/userinfo"
-    else
-      ENV0_BOUND_AUDIENCE="https://dev.env0.com"
-      AUTH0_BOUND_AUDIENCE="https://dev-env0.auth0.com/userinfo"
-    fi
+echo "Logging in to Vault"
 
-  ./vault write auth/jwt/role/"${VAULT_ROLE}" - <<EOF
-{
-  "user_claim": "sub",
-  "role_type": "jwt",
-  "bound_audiences": [
-    "$ENV0_BOUND_AUDIENCE",
-    "$AUTH0_BOUND_AUDIENCE"
-  ],
-  "bound_claims": {
-    "https://env0.com/organization": "$ENV0_ORGANIZATION_ID",
-    "https://env0.com/apiKeyType": "oidc"
-  }
-}
-EOF
-echo "================="
-fi
+export VAULT_TOKEN=$(./vault write auth/env0-jwt/login role="${VAULT_ROLE}" jwt="${ENV0_OIDC_TOKEN}" -format=json | jq --raw-output '.auth.client_token')
 
+echo "Running some vault commands"
```

Comment:
Do those commands have any meaning? or just running it to make sure it works?

Reply:
no special meaning, they just `put` and `get` a secret from vault. We're only testing authorization here so the content of the commands isn't really important.